config/federation/prometheus/prometheus.yml

```diff
@@ -373,6 +373,42 @@ scrape_configs:
         target_label: __address__
         replacement: blackbox-public-service.default.svc.cluster.local:9115
 
+  # SNMP configurations.
+  - job_name: 'snmp-targets'
+    metrics_path: /snmp
+
+    file_sd_configs:
+      - files:
+          - /snmp-targets/*.json
+        # Attempt to re-read files every five minutes.
+        refresh_interval: 5m
+
+    # This relabel config is necessary. The relabel config redefines the address
+    # to scrape and sets the correct parameters to pass to the scrape target.
+    relabel_configs:
+
+      # The default __address__ value is a target host from the config file.
+      # Here, we set (i.e. "replace") a request parameter "target" equal to the
+      # host value.
+      - source_labels: [__address__]
+        regex: (.*)
```

Comment:
Sorry. I was unclear. What I think will happen is that now prometheus will add `:80` to the end of the address and the value of `__param_target` will be the site switch hostname with port 80 at the end, e.g. `s1.lga1t.measurement-lab.org:80`. When I said "prometheus helpfully adds" the port I meant the "please don't help me" kind of helpful.  Though, this is from memory, so I could be mistaken.

Reply:
I found [some random stackoverflow thread](https://stackoverflow.com/questions/40020053/how-to-relabel-address-with-its-current-value-and-a-label) that somewhat makes it *seem* like Prometheus won't automatically append `:80` to the `__address__` label when interprets `target` from the config. I guess I'll leave it like it is, and if it doesn't work then the fix will be easy.